TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/RequestSenderOrchestrator.java

```diff
@@ -300,27 +366,74 @@ private Duration getDelayFromNowMs(Instant to) {
         return Duration.ofMillis(Math.max(0, Duration.between(now(), to).toMillis()));
     }
 
-    private TrackedFuture<String, AggregatedRawResponse> sendSendingRestOfPackets(
+    private Duration doubleRetryDelayCapped(Duration d) {
+        return Duration.ofMillis(Math.min(d.multipliedBy(2).toMillis(), maxRetryDelay.toMillis()));
+    }
+
+    private <T> TrackedFuture<String, T>
+    sendRequestWithRetries(Supplier<IPacketFinalizingConsumer<AggregatedRawResponse>> senderSupplier,
+                           EventLoop eventLoop,
+                           ByteBufList byteBufList,
+                           Instant referenceStartTime,
+                           Duration nextRetryDelay,
+                           Duration interval,
+                           RetryVisitor<T> visitor)
+    {
+        if (eventLoop.isShuttingDown()) {
+            return TextTrackedFuture.failedFuture(new IllegalStateException("EventLoop is shutting down"),
+                () -> "sendRequestWithRetries is failing due to the pending shutdown of the EventLoop");
+        }
+        return sendPackets(senderSupplier.get(), eventLoop,
+            byteBufList.streamRetained().iterator(), referenceStartTime, interval, new AtomicInteger())
```

Comment:
I'd prefer this to use refSafeMap to retain the stream and ensure a stream close at the end of the future